### ISSUE TITLE
Increase length of allowed comment

### ISSUE THREAD

**Issue by jonespm** [sakaiproject/sakai#4357]
In GradebookNgBusinessService.java (and EditGradeCommentPanel.java) the comment field is limited to 500 characters and it prints an error if it's too long. In the database, the datatype is clob/longtext. Other tools that link into the gradebook don't seem to have any limits. Maybe unlimited is too much but at least it seems like a good idea to increase it up to 4000, which is when Oracle really changes the data internally to a clob.

**Comment by steveswinsburg** [comment]
IIRC comments are pulled back in the GradebookPage for the entire matrix, so this could be an enormous set of data. I think changing the use case for comments would be a better approach: 
"Good work" rather than some essay.

**Comment by steveswinsburg** [comment]
This is what 4000 characters looks like. Far too much for this field IMO.

> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec pulvinar sem sed mi iaculis bibendum. Nulla convallis nunc nec nisi mollis, sit amet euismod odio varius. Donec quis tortor ut felis interdum interdum. Suspendisse ac velit felis. Duis pretium, leo ac blandit suscipit, mauris nisl tincidunt ante, id pharetra urna nisl eget nulla. Proin scelerisque quis dui id euismod. Nullam sed dictum odio, vel congue eros. Integer risus arcu, tincidunt vitae ex quis, finibus vehicula nisl. Nullam leo arcu, viverra vel sem vitae, molestie semper elit. Sed sollicitudin finibus tincidunt. Nulla malesuada pulvinar viverra. Phasellus a dui et dui pretium vestibulum non id tellus. Fusce in malesuada erat.
> 
> Nullam tincidunt magna et neque pretium, quis elementum turpis egestas. Vivamus at orci ultricies, maximus diam vitae, vehicula elit. Nulla at odio elit. Nam a dignissim felis. Interdum et malesuada fames ac ante ipsum primis in faucibus. Aenean sagittis ultrices velit, a euismod neque ultricies sit amet. Praesent ac nisl pulvinar, venenatis nunc at, tempus nisl. Integer sollicitudin vestibulum urna eu blandit. Vivamus vel faucibus ante, eget bibendum eros. Nulla dignissim ut mauris in elementum.
> 
> Duis vel venenatis augue, pretium ornare dui. Quisque bibendum sed velit at sagittis. Integer porta dolor eu dui mattis, eget mollis odio tristique. Proin quis tristique sem. Aliquam pharetra ex tellus, sit amet ullamcorper nisl varius ac. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Donec ante augue, laoreet vel commodo vel, pretium non diam.
> 
> Pellentesque nunc nisl, pretium nec venenatis sed, auctor eu mi. Aliquam in mi eget ipsum consequat elementum eu sed sem. Mauris vitae enim interdum, euismod turpis at, venenatis sapien. Duis sed laoreet nibh. Ut blandit id massa nec porttitor. Ut a tortor id ex faucibus tincidunt. Donec at fringilla enim. Cras finibus nisl varius lorem bibendum porttitor. Cras rhoncus neque vel quam suscipit pulvinar. Nulla condimentum pellentesque tortor ac efficitur. Vivamus a convallis neque, ut viverra nunc. Nam facilisis semper sem non commodo. Aliquam rutrum enim ut diam ultricies tincidunt. Fusce nulla mauris, eleifend ac tempor et, scelerisque sit amet lorem. Aenean volutpat dui augue, quis lobortis sem egestas et.
> 
> Nam justo ligula, egestas sed justo vel, sodales faucibus mi. Fusce eleifend mollis arcu. Proin ut enim viverra, pretium odio a, viverra ipsum. Sed fringilla elementum tellus id mollis. Donec vel mattis enim. Fusce eu purus nisi. Maecenas blandit eros sit amet velit suscipit, sed pulvinar tellus blandit. Sed nec iaculis mi. Pellentesque malesuada sed enim a efficitur. Nam ac purus lorem. Nam auctor dui mattis nisi tincidunt tristique. Donec semper diam at tempor finibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Vivamus sed leo nec lectus tempus malesuada. Curabitur sagittis mi ac nisl accumsan interdum. Fusce mi lorem, sodales id rutrum in, luctus eu felis.
> 
> Mauris ac odio dapibus, vehicula enim fermentum, vestibulum nisl. Vivamus lorem libero, porta a semper ac, pellentesque ut mi. Donec scelerisque molestie mi. Cras facilisis maximus lacus a tempor. Suspendisse fringilla nunc viverra lacus aliquam facilisis. Curabitur aliquam pulvinar luctus. In posuere arcu non libero aliquet, non convallis risus malesuada. Sed non nisl nec justo varius fermentum ut in orci.
> 
> Nulla non ex sed enim suscipit pretium. In vestibulum sit amet nulla ut sagittis. Maecenas mollis rhoncus nisi ac malesuada. Praesent sed sapien sit amet tortor fermentum auctor. Vestibulum tincidunt purus non arcu semper feugiat. Phasellus interdum dictum finibus. Morbi mollis massa eu tellus hendrerit luctus. Vivamus varius risus non tellus placerat lobortis. Ut consequat pretium feugiat. Etiam viverra sed felis at semper. Suspendisse orci arcu, tincidunt iaculis rutrum vel, elementum iaculis ante. Vivamus felis turpis, imperdiet hendrerit cras amet.

**Comment by jonespm** [comment]
Well, we can't tell instructors how to grade. ;) And maybe 4000 is too much, but 500 is too little. This was a comment from a client that had previously used longer feedback. Maybe it can only show the first 500 characters in the matrix but then have some link that would modal the full comment if it were over that limit? It just seems like when it gets to the GB from other tools it shouldn't be truncated.

**Comment by steveswinsburg** [comment]
We can tell them to change their workflow if they don't have the capability anymore and previous behaviour will now bring an application server down. The modal window is probably an ok approach though.

**Comment by jonespm** [comment]
Yeah we could limit the actual gradebook tool, but still external tools both Sakai and LTI that scored into gradebook, would lose part of the comments. And any existing comments over 500 would also be truncated. I wouldn't put a high priority on fixing this the best way, but we might have to increase it a little for a client locally if they want. Thanks!
